### ISSUE TITLE
Upgrade Python

### ISSUE THREAD

**Issue by leedenison** [hypothesis/product-backlog#1464]
Tasks:
- [x] [Upgrade LMS to a newer version of Python before Oct 2024](https://github.com/hypothesis/lms/issues/5420)
- [x] [Upgrade Checkmate to a newer version of Python before Oct 2024](https://github.com/hypothesis/checkmate/issues/675)
- [x] [Upgrade h to a newer version of Python before Oct 2024](https://github.com/hypothesis/h/issues/7971)
- [x] [Upgrade bouncer to a newer version of Python before Oct 2024](https://github.com/hypothesis/bouncer/issues/643)
    

~~Maybe blocked by: https://github.com/hypothesis/bouncer/pull/705~~

- [x] [Upgrade Via to a newer version of Python before Oct 2024](https://github.com/hypothesis/via/issues/934)
- [x] [Upgrade Via HTML to a newer version of Python before Oct 2024](https://github.com/hypothesis/viahtml/issues/505)

Most likely blocked by https://github.com/webrecorder/pywb/pull/839

- [x] [Upgrade report to a newer version of Python before Oct 2024](https://github.com/hypothesis/report/issues/214)
- [x] [Upgrade h-periodic to a newer version of Python before Oct 2024](https://github.com/hypothesis/h-periodic/issues/308)

## Cleanup

- [ ] [Remove outdated filterwarnings from the cookie cutter](https://github.com/hypothesis/lms/pull/5941/files#r1437064557)

**Comment by marcospri** [comment]
https://github.com/hypothesis/lms/pull/5639

**Comment by robertknight** [comment]
I think this is basically done. viahtml is blocked on pywb support for Python 3.11 I believe, so I'll keep that as a separate issue.
